File: 2_Creational_Design_Patterns/2.02_Factory/Java_Snippet/Factory.java
// Interface representing a Car
interface Car {
    void drive();
}

// Sedan implementation of the Car interface
class Sedan implements Car {
    @Override
    public void drive() {
        System.out.println("Driving Sedan");
    }
}

// SUV implementation of the Car interface
class SUV implements Car {
    @Override
    public void drive() {
        System.out.println("Driving SUV");
    }
}

// Car factory class
class CarFactory {

    // Factory method to create a Car object based on type
    public Car getCar(String carType) {
        if (carType.equals("Sedan")) {
            return new Sedan();
        } else if (carType.equals("SUV")) {
            return new SUV();
        } else {
            throw new IllegalArgumentException("Invalid car type");
        }
    }
}

public class Factory {
    public static void main(String[] args) {
        CarFactory factory = new CarFactory();
        Car car1 = factory.getCar("Sedan");
        car1.drive(); // Output: Driving Sedan

        Car car2 = factory.getCar("SUV");
        car2.drive(); // Output: Driving SUV
    }
}

File: 2_Creational_Design_Patterns/2.02_Factory/Python_Snippet/Factory.py
from abc import ABC, abstractmethod

class Car(ABC):
    @abstractmethod
    def drive(self):
        pass

class Sedan(Car):
    def drive(self):
        print("Driving Sedan")

class SUV(Car):
    def drive(self):
        print("Driving SUV")

class CarFactory:
    def get_car(self, car_type):
        if car_type == "Sedan":
            return Sedan()
        elif car_type == "SUV":
            return SUV()
        else:
            raise ValueError("Invalid car type")

factory = CarFactory()
car1 = factory.get_car("Sedan")
car1.drive()  # Output: Driving Sedan

car2 = factory.get_car("SUV")
car2.drive()  # Output: Driving SUV